### PR TITLE
Replace use of "deployment" with "ndc metadata"

### PR DESCRIPTION
**Pull request by i-am-tom**
I (specifically Tom) keep getting confused about whether deployments are for engine or the NDC bit, and the first time I drafted this PR @plcplc actually had to point out that I got the difference wrong _again_, so this PR replaces mentions of "deployment" with "ndc metadata" to make it more obvious to me (specifically Tom).

### CODE REVIEW

File: scripts/archive-old-deployment.sh
Comment (i-am-tom) [inline scripts/archive-old-deployment.sh:1]: This is now called scripts/archive-old-ndc-metadata.sh

File: crates/tests/tests-common/src/deployment/configuration.rs
Comment (i-am-tom) [inline crates/tests/tests-common/src/deployment/configuration.rs:1]: This is now called crates/tests/tests-common/src/ndc_metadata/configuration.rs

File: crates/tests/tests-common/src/deployment/mod.rs
Comment (i-am-tom) [inline crates/tests/tests-common/src/deployment/mod.rs:1]: This is now called crates/tests/tests-common/src/ndc_metadata/mod.rs

File: crates/tests/databases-tests/src/deployment_snapshot_tests.rs
Comment (i-am-tom) [inline crates/tests/databases-tests/src/deployment_snapshot_tests.rs:1]: This is now called crates/tests/databases-tests/src/ndc_metadata_snapshot_tests.rs

**Review by SamirTalwar — APPROVED** [review]

**Pull request merged**